src/lib/builtins.pl

```diff
@@ -123,24 +205,38 @@
 
 % control operators.
 
+%% fail.
+%
+% A predicate that always fails
 fail :- '$fail'.
 
 
 :- meta_predicate \+(0).
 
+%% \+(Goal)
+%
+% Succeeds if Goal fails
```

Comment:
Ideally, all instances of *if* should become *iff* (or spelt out: *if and only if*) when they can be strengthened like this. Also, it is usually better to talk about what the predicates *mean* instead of what they *do*, because the meaning is often more general and can be applied to different instantiation modes. For instance, in this case, the wording could become:

*True iff Goal fails.*